bifrost/inmemory_locker.go

```diff
@@ -0,0 +1,95 @@
+package bifrost
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Lock struct {
+	lockPath string
+	uuid     string
+	reason   string
+}
+
+type InMemoryLocker struct {
+	locks map[string][]*Lock
+
+	mx sync.Mutex
+}
+
+func NewInMemoryLocker() *InMemoryLocker {
+	return &InMemoryLocker{}
+}
+
+func (l *InMemoryLocker) init() {
+	if l.locks == nil {
+		l.locks = map[string][]*Lock{}
+	}
```

Comment:
Is there any reason to have allocation and initialization as two separate functions? Combining them here allows you to remove multiple `init` calls below.

```suggestion
	return &InMemoryLocker{locks: make(map[string][]*Lock)}
```

Reply:
No real reason. Looking at it again, I believe it was because I initially built a mock DynamoDB client based on the mock S3 client which used the `init()` pattern. I forgotten to clean it up after creating this implementation, and after pulling it out of the test.

That said, I did go for the `New<Name>` pattern based on what was already being done elsewhere (e.g. Pro deployer). Though, generally not a fan of it versus initialising using the struct directly.